packages/utils/src/timeout.ts

```diff
@@ -1,23 +1,35 @@
-import {anySignal} from "any-signal";
-import {TimeoutError} from "./errors.js";
-import {sleep} from "./sleep.js";
+import { TimeoutError } from "./errors.js";
+import { sleep } from "./sleep.js";
 
 export async function withTimeout<T>(
   asyncFn: (timeoutAndParentSignal?: AbortSignal) => Promise<T>,
   timeoutMs: number,
   signal?: AbortSignal
 ): Promise<T> {
   const timeoutAbortController = new AbortController();
-  const timeoutAndParentSignal = anySignal([timeoutAbortController.signal, ...(signal ? [signal] : [])]);
+  const timeoutAndParentSignal = timeoutAbortController.signal;
+
+  let abortListener: (() => void) | undefined;
+
+  if (signal) {
+    abortListener = () => timeoutAbortController.abort();
+    signal.addEventListener("abort", abortListener);
+  }
 
   async function timeoutPromise(signal: AbortSignal): Promise<never> {
     await sleep(timeoutMs, signal);
     throw new TimeoutError();
   }
 
   try {
-    return await Promise.race([asyncFn(timeoutAndParentSignal), timeoutPromise(timeoutAndParentSignal)]);
+    return await Promise.race([
+      asyncFn(timeoutAndParentSignal),
+      timeoutPromise(timeoutAndParentSignal),
+    ]);
   } finally {
     timeoutAbortController.abort();
```

Comment:
The `finally` block always execute before control flow exits the block. So `finally` could be triggered in two cases: 

1. If there is any error in executing `asyncFn` 
2. If process timeout 

In both cases we should not trigger the `.abort()` on this controller.